applications/plugins/Communication/components/serverCommunicationQSerial.inl

```diff
@@ -0,0 +1,330 @@
+        int nbCols  = typeinfo->size(data->getValueVoidPtr()) / typeinfo->size();
+        messageStr << "matrixint:" << std::to_string(nbRows) << " QSerialint:" << std::to_string(nbCols) << " ";
+
+        if( !typeinfo->Text() && !typeinfo->Scalar() && !typeinfo->Integer() )
+        {
+            msg_advice(data->getOwner()) << "BaseData_getAttr_value unsupported native type="<<data->getValueTypeString()<<" for data "<<data->getName()<<" ; returning string value" ;
+            messageStr << "QSerialstring:'" << (data->getValueString()) << "' ";
+        }
+        else if (typeinfo->Text())
+            for (int i=0; i < nbRows; i++)
+                for (int j=0; j<nbCols; j++)
+                    messageStr << "QSerialstring:" << typeinfo->getTextValue(valueVoidPtr,(i*nbCols) + j).c_str() << " ";
+        else if (typeinfo->Scalar())
+            for (int i=0; i < nbRows; i++)
+                for (int j=0; j<nbCols; j++)
+                    messageStr << "QSerialfloat:" << (float)typeinfo->getScalarValue(valueVoidPtr,(i*nbCols) + j) << " ";
+        else if (typeinfo->Integer())
+            for (int i=0; i < nbRows; i++)
+                for (int j=0; j<nbCols; j++)
+                    messageStr << "QSerialint:" << (int)typeinfo->getIntegerValue(valueVoidPtr,(i*nbCols) + j) << " ";
+    }
+    else
+    {
+        if( !typeinfo->Text() && !typeinfo->Scalar() && !typeinfo->Integer() )
+        {
+            msg_advice(data->getOwner()) << "BaseData_getAttr_value unsupported native type=" << data->getValueTypeString() << " for data "<<data->getName()<<" ; returning string value" ;
+            messageStr << "QSerialstring:'" << (data->getValueString()) << "' ";
+        }
+        if (typeinfo->Text())
+        {
+            messageStr << "QSerialstring:'" << (data->getValueString()) << "' ";
+        }
+        else if (typeinfo->Scalar())
+        {
+            messageStr << "QSerialfloat:" << (data->getValueString()) << " ";
+        }
+        else if (typeinfo->Integer())
+        {
+            messageStr << "QSerialint:" << (data->getValueString()) << " ";
+        }
+    }
+    delete data;
+    return messageStr.str();
+}
+
+/******************************************************************************
+*                                                                             *
+* RECEIVE PART                                                                *
+*                                                                             *
+******************************************************************************/
+
+void ServerCommunicationQSerial::receiveData()
+{
+    std::string port = d_port.getValueString();
+    while(m_running)
+    {
+        serial->setPort(port);
+        serial->setBaudRate(QSerialPort::Baud9600);
+        serial->setDataBits(QSerialPort::Data8);
+        if (serial->open(QIODevice::ReadOnly))
+        {
+            serial->setDataTerminalReady(true);
```

Comment:
open it only one time then loop on it.

Reply:
Can we skip this part because I will not be able to test it, even if I make suitable changes?

Reply:
you are not able to test it but it please, do the changes :)